app/src/main/java/org/simple/clinic/enterotp/EnterOtpUpdate.kt

```diff
@@ -30,11 +30,18 @@ class EnterOtpUpdate(
 
   private fun effectsForStateChanged(stateChanged: ProtectedState): Next<EnterOtpModel, EnterOtpEffect> {
     return when (stateChanged) {
-      is Allowed -> dispatch(AllowOtpEntry)
+      is Allowed -> dispatch(generateEffectForAllowingOtpEntry(stateChanged), AllowOtpEntry)
       else -> // do something
     }
   }
 
+  private fun generateEffectForAllowingOtpEntry(protectedState: Allowed): EnterOtpEffect {
+    return if (protectedState.attemptsMade == 0) {
+      HideErrors
```

Comment:
Usually, we just return `HideErrors` directly and in later commits we add the `if-else` condition.

Reply:
Ah I see, noted boss